### ISSUE TITLE
SNMP Traps from Juniper not fully translated

### ISSUE THREAD

**Issue by hkml2000** [influxdata/telegraf#10309]
### Relevent telegraf.conf

```toml
[[inputs.snmp_trap]]
  service_address = "udp://:162"
  path = ["/usr/share/snmp/mibs"]
  timeout = "10s"
  version = "2c"

[[outputs.file]]
  files = ["stdout"]
  data_format = "influx"
```

### System info

Telegraf 1.21.1

### Steps to reproduce

1. put Juniper MIB files into /usr/share/snmp/mibs

2. start telegraf using above config

3. send snmptrap  using "snmptrap -v 2c -c public **telegraf_server_ip** '' .1.3.6.1.4.1.2636.3.53.1.2.1.1"

4. observe the output from telegraf :  
snmp_trap,community=public,host=xxxxxx,mib=**JUNIPER-SMI**, name=**jnxL2cpMibRoot**,oid=.1.3.6.1.4.1.2636.3.53.1.2.1.1,source=x.x.x.x,version=2c sysUpTime=130777096i 1640015154364309647
translation don't go beyond JUNIPER-SMI MIB

5. using snmptranslate command  with the same OID :  
snmptranslate 1.3.6.1.4.1.2636.3.53.1.2.1.1 
==> we obtain: JUNIPER-L2CP-FEATURES-MIB::**jnxLacpTimeOut**

### Expected behavior

load all MIBs to fully translate snmp traps

### Actual behavior

partial snmp trap translation, seems like not all MIBs from path are loaded

**Comment by Hipska** [comment]
Problem is similar to #9470, but the root cause is different. Here it seems because for some reason the MIB cannot be parsed, on the other issue the MIB is just missing.

Could you check if you have any mib parsing error messages during startup of telegraf?

**Comment by hkml2000** [comment]
> Problem is similar to #9470, but the root cause is different. Here it seems because for some reason the MIB cannot be parsed, on the other issue the MIB is just missing.
> 
> Could you check if you have any mib parsing error messages during startup of telegraf?

I turned on the debug and the output shows no error:

2021-12-21T09:02:15Z D! [agent] Connecting outputs
2021-12-21T09:02:15Z D! [agent] Attempting connection to [outputs.influxdb_v2]
2021-12-21T09:02:15Z D! [agent] Successfully connected to outputs.influxdb_v2
2021-12-21T09:02:15Z D! [agent] Starting service inputs
2021-12-21T09:02:15Z I! [inputs.snmp_trap] Listening on udp://:162
2021-12-21T09:02:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:02:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:02:45Z D! [outputs.influxdb_v2] Wrote batch of 7 metrics in 13.511724ms
2021-12-21T09:02:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:02:55Z D! [outputs.influxdb_v2] Wrote batch of 5 metrics in 5.733307ms
2021-12-21T09:02:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:03:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:03:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:03:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:03:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:03:45Z D! [outputs.influxdb_v2] Wrote batch of 7 metrics in 43.47519ms
2021-12-21T09:03:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:03:55Z D! [outputs.influxdb_v2] Wrote batch of 1 metrics in 5.626479ms
2021-12-21T09:03:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:04:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:04:15Z D! [outputs.influxdb_v2] Wrote batch of 1 metrics in 9.688627ms
2021-12-21T09:04:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:04:25Z D! [outputs.influxdb_v2] Wrote batch of 4 metrics in 6.15422ms
2021-12-21T09:04:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:04:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:04:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:04:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:05:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:05:15Z D! [outputs.influxdb_v2] Wrote batch of 4 metrics in 8.226266ms
2021-12-21T09:05:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:05:25Z D! [outputs.influxdb_v2] Wrote batch of 4 metrics in 10.554273ms
2021-12-21T09:05:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:05:35Z D! [outputs.influxdb_v2] Wrote batch of 4 metrics in 8.212175ms
2021-12-21T09:05:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:05:45Z D! [outputs.influxdb_v2] Wrote batch of 5 metrics in 7.649121ms
2021-12-21T09:05:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:05:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:06:05Z D! [outputs.influxdb_v2] Wrote batch of 37 metrics in 12.389268ms
2021-12-21T09:06:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:06:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:06:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:06:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:06:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:06:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:07:05Z D! [outputs.influxdb_v2] Wrote batch of 4 metrics in 5.379493ms
2021-12-21T09:07:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:07:15Z D! [outputs.influxdb_v2] Wrote batch of 11 metrics in 6.250572ms
2021-12-21T09:07:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:07:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:07:35Z D! [outputs.influxdb_v2] Wrote batch of 3 metrics in 5.462284ms
2021-12-21T09:07:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:07:45Z D! [outputs.influxdb_v2] Wrote batch of 1 metrics in 5.966987ms
2021-12-21T09:07:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:07:55Z D! [outputs.influxdb_v2] Wrote batch of 8 metrics in 8.204441ms
2021-12-21T09:07:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:08:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:08:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:08:25Z D! [outputs.influxdb_v2] Wrote batch of 3 metrics in 7.356136ms
2021-12-21T09:08:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:08:35Z D! [outputs.influxdb_v2] Wrote batch of 5 metrics in 7.332634ms
2021-12-21T09:08:35Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:08:45Z D! [outputs.influxdb_v2] Wrote batch of 1 metrics in 8.188778ms
2021-12-21T09:08:45Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:08:55Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:09:05Z D! [outputs.influxdb_v2] Wrote batch of 8 metrics in 7.205404ms
2021-12-21T09:09:05Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:09:15Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
2021-12-21T09:09:25Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics

**Comment by Hipska** [comment]
And you are sure the permissions for `JUNIPER-L2CP-FEATURES-MIB` are ok for the telegraf user, and it exists in the same directory?

**Comment by Hipska** [comment]
There should also be some output before your first line. The place where the errors would be visible is left out :p

**Comment by hkml2000** [comment]
mib files exist in the same directory with read permission 

here is the full output, no error appears:

telegraf.service - The plugin-driven server agent for reporting metrics into InfluxDB
   Loaded: loaded (/usr/lib/systemd/system/telegraf.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2021-12-22 16:12:54 CET; 37s ago
     Docs: https://github.com/influxdata/telegraf
 Main PID: 27148 (telegraf)
   CGroup: /system.slice/telegraf.service
           â””â”€27148 /usr/bin/telegraf -config /etc/telegraf/telegraf.conf -config-directory /etc/telegraf/telegraf.d

Dec 22 16:12:54 telegraf telegraf[27148]: 2021-12-22T15:12:54Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"telegraf", Flush Interval:10s
Dec 22 16:12:54 telegraf telegraf[27148]: 2021-12-22T15:12:54Z D! [agent] Initializing plugins
Dec 22 16:13:09 telegraf telegraf[27148]: 2021-12-22T15:13:09Z D! [agent] Connecting outputs
Dec 22 16:13:09 telegraf telegraf[27148]: 2021-12-22T15:13:09Z D! [agent] Attempting connection to [outputs.influxdb_v2]
Dec 22 16:13:09 telegraf telegraf[27148]: 2021-12-22T15:13:09Z D! [agent] Successfully connected to outputs.influxdb_v2
Dec 22 16:13:09 telegraf telegraf[27148]: 2021-12-22T15:13:09Z D! [agent] Starting service inputs
Dec 22 16:13:09 telegraf telegraf[27148]: 2021-12-22T15:13:09Z I! [inputs.snmp_trap] Listening on udp://:162
Dec 22 16:13:19 telegraf telegraf[27148]: 2021-12-22T15:13:19Z D! [outputs.influxdb_v2] Wrote batch of 12 metrics in 11.27086ms
Dec 22 16:13:19 telegraf telegraf[27148]: 2021-12-22T15:13:19Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics
Dec 22 16:13:29 telegraf telegraf[27148]: 2021-12-22T15:13:29Z D! [outputs.influxdb_v2] Buffer fullness: 0 / 20000 metrics

**Comment by hkml2000** [comment]
just to add that with previous release of telegraf like 1.14 the translation works fine

**Comment by MyaLongmire** [comment]
Can you please post the mib file and I will try to figure out why this happening?

**Comment by Hipska** [comment]
Hi, could you test with telegraf 1.21.3?

**Comment by hkml2000** [comment]
Hi,

after upgrade to 1.21.3,  telegraf starts normaly, plugins are loaded but no listening on port udp 162 is reported in the logs
no traps are received.

after rollback to 1.21.1, I realize that snmp translation works fine !

**Comment by Hipska** [comment]
Yes, I see the same behaviour, logged #10548 for that.

So am I right the translation problem mentioned in this issue is solved in 1.21.1?

**Comment by hkml2000** [comment]
yes, it's solved in 1.21.1
